common/patches/rof2.cpp

```diff
@@ -5076,38 +5086,69 @@ namespace RoF2
 	DECODE(OP_RaidInvite)
 	{
 		DECODE_LENGTH_ATLEAST(structs::RaidGeneral_Struct);
+		RaidGeneral_Struct* rgs = (RaidGeneral_Struct*)__packet->pBuffer;
 
-		// This is a switch on the RaidGeneral action
-		switch (*(uint32 *)__packet->pBuffer) {
-			case 35: { // raidMOTD
-				// we don't have a nice macro for this
-				structs::RaidMOTD_Struct *__eq_buffer = (structs::RaidMOTD_Struct *)__packet->pBuffer;
-				__eq_buffer->motd[1023] = '\0';
-				size_t motd_size = strlen(__eq_buffer->motd) + 1;
-				__packet->size = sizeof(RaidMOTD_Struct) + motd_size;
-				__packet->pBuffer = new unsigned char[__packet->size];
-				RaidMOTD_Struct *emu = (RaidMOTD_Struct *)__packet->pBuffer;
-				structs::RaidMOTD_Struct *eq = (structs::RaidMOTD_Struct *)__eq_buffer;
-				strn0cpy(emu->general.player_name, eq->general.player_name, 64);
-				strn0cpy(emu->motd, eq->motd, motd_size);
-				IN(general.action);
-				IN(general.parameter);
-				FINISH_DIRECT_DECODE();
-				break;
-			 }
-			case 36: { // raidPlayerNote unhandled
-				break;
-			}
-			default: {
-				DECODE_LENGTH_EXACT(structs::RaidGeneral_Struct);
-				SETUP_DIRECT_DECODE(RaidGeneral_Struct, structs::RaidGeneral_Struct);
-				strn0cpy(emu->leader_name, eq->leader_name, 64);
-				strn0cpy(emu->player_name, eq->player_name, 64);
-				IN(action);
-				IN(parameter);
-				FINISH_DIRECT_DECODE();
-				break;
-			}
+		switch (rgs->action)
+		{
+		case 35: {
+			//Start Here.  Setup the buffer into a temp location and map to eq
+			unsigned char* in = __packet->pBuffer;
+			structs::RaidMOTD_Struct* eq = (structs::RaidMOTD_Struct*)in;
+
+			//Setup a new object using the new Struct
+			RaidMOTD_Struct* emu = new RaidMOTD_Struct();
+
+			//Map values and convert char to string to be used within source
+			IN(general.action);
+			IN(general.parameter);
+			IN_str(general.leader_name);
+			IN_str(general.player_name);
+			emu->motd = std::string(eq->motd);
+
+			//Point buffer to new structure
+			__packet->size = sizeof(RaidNote_Struct);
+			__packet->pBuffer = (unsigned char*)emu;
```

Comment:
This will cause the `std::string` to leak in the packet struct because `RaidMOTD_Struct` never destructs.

The `std::string` in `RaidMOTD_Struct` makes it non-trivially destructible. When `pBuffer` is deleted later the `RaidMOTD_Struct` will not destruct (just its memory is deleted) which will result in the memory allocated by the std::string to leak. The `RaidMOTD_Struct` would have to be deleted here to destruct it properly, but the only way to do that is by memcpying to the buffer which would also be undefined behavior because of the `std::string`.